src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -142,12 +142,18 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
           }
           break;
         case FORMAT_NAME_NODES :
-          if (!liveState.isNameNode1Initialized()) {
+          if (!liveState.isNameNode1Initialized() && !liveState.isNameNode2Initialized()) {
             sendMessageTo(
                 driver,
                 liveState.getFirstNameNodeTaskId(),
                 liveState.getFirstNameNodeSlaveId(),
                 HDFSConstants.NAME_NODE_INIT_MESSAGE);
+          } else if (!liveState.isNameNode1Initialized()) {
```

Comment:
The first condition (line 145) checks if neither of the NNs has been initialized/bootstrapped, in which case we need to initialize one of them (NN1).
The second condition (line 151) then tests if NN2 /is/ initialized/bootstrapped, but NN1 is (no longer) initialized. This could happen if NN1 fails after both have already been initialized. In this case, since NN2 has the valid metadata (and is presumably the new active NN), we want to bootstrap NN1 off of NN2.